working/0546-patterns/patterns-feature-specification.md

```diff
@@ -541,24 +540,40 @@ is expected.
 ### List pattern
 
 ```
-listPattern ::= typeArguments? '[' patterns? ']'
+listPattern         ::= typeArguments? '[' listPatternElements? ']'
+listPatternElements ::= listPatternElement ( ',' listPatternElement )* ','?
+listPatternElement  ::= pattern | restPattern
+restPattern         ::= '...' pattern?
 ```
 
 A list pattern matches an object that implements `List` and extracts elements by
 position from it.
 
-It is a compile-time error if `typeArguments` is present and has more than one
-type argument.
+It is a compile-time error if:
+
+*   `typeArguments` is present and has more than one type argument.
+
+*   There is more than one `restPattern` element in the list pattern. *It can
+    appear anywhere in the list, but there can only be zero or one.*
 
-**TODO: Allow a `...` element in order to match suffixes or ignore extra
-elements. Allow capturing the rest in a variable.**
+#### Rest patterns
+
+A list pattern may contain a *rest element* which allows matching lists of
+arbitrary lengths. If a rest element is present and has a subpattern, all of the
+elements not matched by other subpatterns are collected into a new list and that
```

Comment:
What is the runtime type of the new list?

Reply:
I think you answer this later implicitly, via the specification of this as a call to `sublist`, right?

Reply:
Yeah, the runtime type falls out of that.